frontend/javascripts/dashboard/dataset/queries.tsx

```diff
@@ -60,8 +65,13 @@ export function useDatasetSearchQuery(query: string | null) {
   );
 }
 
-export function useFolderTreeQuery() {
-  return useQuery(["folders"], getFolderTree, {
+async function fetchTreeHierarchy() {
+  const flatTreeItems = await getFolderTree();
+  return getFolderHierarchy(flatTreeItems);
```

Comment:
The query stores the hierarchical data instead of the flat backend data now. This is also why the (optimistic) updates in the queries had to be changed (they were only adapted to use the new helper `transformHierarchy` so that the old update code can be reused 1:1).